### PR TITLE
feat(Nebula): Add nebula.tv support

### PR DESCRIPTION
**Pull request by angelolz**
## Description 
This PR adds a new presence for the streaming site [Nebula](https://nebula.tv/)

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

image

image

</details>

### CODE REVIEW

File: websites/N/Nebula/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"description": {
+		"en": "Nebula is the home of smart, thoughtful videos, podcasts, and classes from your favorite creators."
+	},
+	"url": "nebula.tv",
+	"matches": [
+		"*://nebula.tv/*"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/OzDunh4.png",
+	"thumbnail": "https://i.imgur.com/4wI2vwl.jpeg",
```
Comment (theusaf) [inline websites/N/Nebula/metadata.json:17]: This image URL directs to a 404 page.

File: websites/N/Nebula/presence.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+}
+
+function getVideoDetails(
+	presenceData: PresenceData,
+	showButtons: boolean
+): void {
+	const videoElement = document.querySelector("video");
+
+	presenceData.details = document.querySelector(".css-11wyb0j").innerHTML;
+	presenceData.state = document.querySelector(".css-r06ha9").innerHTML;
```
Comment (theusaf) [inline websites/N/Nebula/presence.ts:124]: Use `textContent` over `innerHTML`.

Comment (theusaf) [inline websites/N/Nebula/presence.ts:124]: If possible, use selectors that don't involve computer-generated class names like these.

File: websites/N/Nebula/presence.ts [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+	setTimestamps(videoElement, presenceData);
+
+	if (showButtons) {
+		presenceData.buttons = [
+			{
+				label: "Watch Video",
+				url: document.location.href,
+			},
+			{
+				label: "View channel",
```
Comment (theusaf) [inline websites/N/Nebula/presence.ts:136]:
```suggestion
				label: "View Channel",
```

File: websites/N/Nebula/presence.ts [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+	} else presenceData.smallImageKey = Assets.Play;
+}
+
+interface QueryParams {
+	[key: string]: string;
+}
+
+function parseQueryParams(): QueryParams {
+	const queryParams: QueryParams = {},
+		queryString = window.location.search.split("?")[1];
```
Comment (theusaf) [inline websites/N/Nebula/presence.ts:241]: Use `document.location` for consistency
```suggestion
		queryString = document.location.search.split("?")[1];
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/N/Nebula/presence.ts (hunk 3 above)
Comment (angelolz) [reply websites/N/Nebula/presence.ts]: I've applied this suggestion to this area and other related areas where button labels are used. Thanks!

**Comment by angelolz** [comment]
I didn't know if I had to be very specific with the selectors I used, but if there's a better way for me to select these elements, please let me know!

File: websites/N/Nebula/presence.ts [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = document.location,
+		[showButtons] = await Promise.all([
+			presence.getSetting<boolean>("buttons"),
+		]),
```
Comment (DarkVillager) [inline websites/N/Nebula/presence.ts:18]: While getting one promise, don't use Promise.all
```suggestion
		showButtons = 
			await presence.getSetting<boolean>("buttons"),
```

File: websites/N/Nebula/presence.ts [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+const enum Assets { // Other default assets can be found at index.d.ts
+	Logo = "https://i.imgur.com/VnwSEyu.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = document.location,
```
Comment (DarkVillager) [inline websites/N/Nebula/presence.ts:15]:
```suggestion
		{ pathname, href } = document.location,
```

File: websites/N/Nebula/presence.ts [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+		presenceData.state = document.querySelector(
+			`${classInfoElementSelector} > div:nth-of-type(3)`
+		).textContent;
+		setTimestamps(videoElement, presenceData);
+
+		if (showButtons) {
+			presenceData.buttons = [
+				{
+					label: "Watch Episode",
+					url: document.location.href,
```
Comment (DarkVillager) [inline websites/N/Nebula/presence.ts:240]:
```suggestion
					url: href,
```

File: websites/N/Nebula/presence.ts [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+			"main > div:nth-of-type(1) > div:nth-of-type(3) > div > div:nth-of-type(2) > div:nth-of-type(1)"
+		).textContent;
+		presenceData.state = channelElement.textContent;
+		setTimestamps(audioElement, presenceData);
+
+		if (showButtons) {
+			presenceData.buttons = [
+				{
+					label: "Listen to Podcast",
+					url: document.location.href,
```
Comment (DarkVillager) [inline websites/N/Nebula/presence.ts:208]:
```suggestion
					url: href,
```

File: websites/N/Nebula/presence.ts [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+		} else {
+			presenceData.details = "Viewing a channel";
+			presenceData.state = channelName.textContent;
+		}
+
+		if (showButtons) {
+			presenceData.buttons = [
+				{
+					label: channelName === null ? "View Podcast" : "View Channel",
+					url: document.location.href,
```
Comment (DarkVillager) [inline websites/N/Nebula/presence.ts:188]:
```suggestion
					url: href,
```

File: websites/N/Nebula/presence.ts [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+		];
+	}
+
+	if (videoElement === null) return;
+	setTimestamps(videoElement, presenceData);
+}
+
+function getSearchDetails(presenceData: PresenceData): void {
+	presenceData.details = "Searching for:";
+	presenceData.state = parseQueryParams().q || "...";
```
Comment (DarkVillager) [inline websites/N/Nebula/presence.ts:155]: Optional, but would look ok
```suggestion
	presenceData.state = parseQueryParams().q || "...";
	presenceData.smallImageKey = Assets.Search;
```

File: websites/N/Nebula/presence.ts [hunk 11]
```diff
@@ -0,0 +1,285 @@
...
+
+function getVideoDetails(
+	presenceData: PresenceData,
+	showButtons: boolean
+): void {
+	const videoElement = document.querySelector("video");
+
+	presenceData.details = document.querySelector(
+		"[aria-label='video description'] > div:nth-of-type(1) > h1"
+	).textContent;
```
Comment (DarkVillager) [inline websites/N/Nebula/presence.ts:125]: Can you shorten these selectors e.g. or does this not work?
```suggestion
	presenceData.details = document.querySelector(
		"[aria-label='video description']"
	).querySelector("h1").textContent;
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/N/Nebula/presence.ts (hunk 11 above)
Comment (angelolz) [reply websites/N/Nebula/presence.ts:125]: Yeah, I think I might've overdone it with the selectors. I've shortened them to the best of my abilities.

File: websites/N/Nebula/metadata.json [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+	"description": {
+		"en": "Nebula is the home of smart, thoughtful videos, podcasts, and classes from your favorite creators."
+	},
+	"url": "nebula.tv",
+	"matches": [
+		"*://nebula.tv/*"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/VnwSEyu.png",
+	"thumbnail": "https://i.imgur.com/fNXHoQg.jpeg",
```
Comment (theusaf) [inline websites/N/Nebula/metadata.json:17]: I'm not sure why, but I get 404 for this. 
However, the previous one (`https://i.imgur.com/4wI2vwl.jpeg`) seems to work now. Perhaps its a caching/dns problem on my side.

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**